third_party/FMI/main.c

```diff
@@ -441,38 +456,38 @@ static int error(const char* message){
 ///\param sizemodelID Size of FMU modelID
 ///\param fmiVersionNumber FMI version number
 ////////////////////////////////////////////////////////////////
-fmiInteger getfmiEPlusVersion(char* fmuWorkingFolder,
-	fmiInteger *sizefmuWorkingFolder, char *fmiVersionNumber, fmiInteger *index){
-
-		char *trimfmuWorkingFolder;
-		fmiString verID;
-		//char* trimmodelID;
-		char* err_msg = "Check FMU binaries folder and see whether libraries "
-			"exist for the system architecture of the EnergyPlus version used. Also check whether the FMU has "
-			"been exported for Co-Simulation. FMU for Model Exchange is not supported yet.";
-
-		// allocate memory for the FMU-working folder trimmed
-		trimfmuWorkingFolder = (char*)calloc(sizeof(char),*sizefmuWorkingFolder + 1);
-		//write fmuWorkingFolder withouth blanks
-		strncpy(trimfmuWorkingFolder, fmuWorkingFolder, *sizefmuWorkingFolder);
-
-		// get index value
-		_c->index = *index;
-
-		//load lib by specifying path to the binaries
-		if (loadLib(trimfmuWorkingFolder, fmuInstances[_c->index]->modelID, fmuInstances[_c->index])) {
-			strncpy(fmiVersionNumber, err_msg, strlen(err_msg));
-			return -1;
-		}
-
-		// free trimmed working folder
-		free (trimfmuWorkingFolder);
-
-		// get the modelID of the FMU
-		verID = fmuInstances[_c->index]->getVersion();
-		strncpy(fmiVersionNumber, verID, strlen (verID)+ 1);
-
-		return 0;
+fmiInteger getfmiEPlusVersion(char *fmuWorkingFolder, fmiInteger *sizefmuWorkingFolder, char *fmiVersionNumber, fmiInteger *index)
+{
+
+    char *trimfmuWorkingFolder;
+    fmiString verID;
+    // char* trimmodelID;
+    char *err_msg = "Check FMU binaries folder and see whether libraries "
+                    "exist for the system architecture of the EnergyPlus version used. Also check whether the FMU has "
+                    "been exported for Co-Simulation. FMU for Model Exchange is not supported yet.";
+
+    // allocate memory for the FMU-working folder trimmed
+    trimfmuWorkingFolder = (char *)calloc(*sizefmuWorkingFolder + 1, sizeof(char));
+    // write fmuWorkingFolder withouth blanks
+    strncpy(trimfmuWorkingFolder, fmuWorkingFolder, *sizefmuWorkingFolder);
+
+    // get index value
+    _c->index = *index;
+
+    // load lib by specifying path to the binaries
+    if (loadLib(trimfmuWorkingFolder, fmuInstances[_c->index]->modelID, fmuInstances[_c->index])) {
+        memcpy(fmiVersionNumber, err_msg, strlen(err_msg));
```

Comment:
`stringop-overflow` warning.  (side note: `fmiVersionNumber` is supposed to be allocated outside of the function, hoping that it is indeed, and correctly...)